compiler.ts

```diff
@@ -81,6 +80,21 @@ export function compile(ast: Program<Type>, env: GlobalEnv) : CompileResult {
 function codeGenStmt(stmt: Stmt<Type>, env: GlobalEnv): Array<string> {
   switch (stmt.tag) {
     case "store":
+      if(stmt.start.tag === "listaddr") {
```

Comment:
I think this isn't the right design. I would expect that all of the work for looking up/storing lists can be done in `lower.ts`. You can use `ifjmp` to manage the errors (or add special functions like `assert_not_none`), and use `alloc` and `store` along with intermediate variables to do this work. This will complicate optimization, memory management, and so on to have special type-specific operations within the code generation step.